launchdarkly-android-client-sdk/src/main/java/com/launchdarkly/sdk/android/LDClient.java

```diff
@@ -530,6 +530,13 @@ private EvaluationDetail<LDValue> variationDetailInternal(@NonNull String key, @
                     null, defaultValue, false, null);
             result = EvaluationDetail.fromValue(defaultValue, EvaluationDetail.NO_VARIATION, EvaluationReason.error(EvaluationReason.ErrorKind.FLAG_NOT_FOUND));
         } else {
+            if (flag.getPrerequisites() != null) {
+                // recurse on prerequisites to emulate prereq evaluations occurring with desirable side effects such as events for prereqs
+                for (String prereqKey : flag.getPrerequisites()) {
+                    variationDetailInternal(prereqKey, LDValue.ofNull(), false, false);
```

Comment:
For reviewers: Is LDValue.ofNull() an appropriate default value for prereq evals?